### ISSUE TITLE
post-import and post-export and some other issues

### ISSUE THREAD

**Issue by netfreak98** [natesales/bcg#11]
Hi,

could you please also implement something like a post import and export configuration option which adds own functions after the "bgp_large_community.add" statement in the filters? In my case I'm trying to filter for specific commuities and replace them but in the end the default community is set my the filter for all prefixes :D 

```
    ipv4 {
        import keep filtered;
        import limit AS49697_IBGP_ROUTE4_REFLECTOR_MAXPFX_v4 action disable;
        next hop self;
        import filter {
            reject_bogons();


            strip_info_communities();
            bgp_local_pref = 100;
            honor_graceful_shutdown();
            AS49697_IBGP_RR_IN();
            bgp_large_community.add((ASN,0,102)); # Learned from peer

            accept;
        };

        export filter {
            AS49697_IBGP_RR_OUT();

            bgp_path.delete([64512..65534, 4200000000..4294967294]); # Strip private ASNs

            accept_local(); # Originated
            process_prepends();
            if ((ASN,0,103) ~ bgp_large_community) then accept; # Downstream

            reject;
        };
    };

```
In the above example "AS49697_IBGP_RR_IN()" and "AS49697_IBGP_RR_OUT()" are my individual functions which are just matching for downstream, updateam and peer large-communites. 

```
function AS49697_IBGP_RR_OUT() {

  if ((ASN,0,101) ~ bgp_large_community) then {
      accept;
    }

  else if ((ASN,0,102) ~ bgp_large_community) then {
      accept;
    }

  else if ((ASN,0,103) ~ bgp_large_community) then {
      accept;
    }

  else reject;
}

function AS49697_IBGP_RR_IN() {

  if ((ASN,0,101) ~ bgp_large_community) then {
      bgp_local_pref = 100;
      accept;
    }

  else if ((ASN,0,102) ~ bgp_large_community) then {
      bgp_local_pref = 200;
      accept;
    }

  else if ((ASN,0,103) ~ bgp_large_community) then {
      bgp_local_pref = 500;
      accept;
    }

  else reject;
}
```

Unfortunately there is also the function "strip_info_communities()" which just removes everything for which I want to match for :D

```
function strip_info_communities() {
  bgp_large_community.delete([(ASN, 0, *)]);
}
```
A workaround would be to use alternative communities but it would make it more complex. How could we make this work? By implementing another peer-type which has a more open filtering or something else?

Thanks!
Joey

**Comment by netfreak98** [comment]
removing strip_info_communities() from the protocol manually fixed it temporary.

**Comment by natesales** [comment]
I've added two new options, `pre-import-final` and `pre-export-final` to add a BIRD config expression right before the final return value (accept or reject). I don't want to add another peer type for this case, but this should work for your use case.

Closed in 769b3962388da91ca9d841e4a0f6b788d58d9656
